### PR TITLE
Cleaning up dependencies

### LINKED ISSUE #178: pyviz tutorial is failing to read .parq files
I'm running the pyviz examples inside a docker container (see #175 ).  Everything is working except thing involve with ready .parq files. I am getting the following output.  Any idea:

Notebook: **01_Workflow_Introduction**
```
import dask.dataframe as dd, geoviews as gv, cartopy.crs as crs
from colorcet import palette
from holoviews.operation.datashader import datashade
from geoviews.tile_sources import EsriImagery

topts = hv.opts.WMTS(width=700, height=600, bgcolor='black', 
                     xaxis=None, yaxis=None, show_grid=False)
tiles = EsriImagery.clone(crs=crs.GOOGLE_MERCATOR).opts(topts) 
taxi  = dd.read_parquet('../data/nyc_taxi_wide.parq').persist()
colormaps = {n: palette[n] for n in ['fire','bgy','bgyw','bmy','gray','kbc']}
```

**Output:**
```
---------------------------------------------------------------------------
ArrowIOError                              Traceback (most recent call last)
<ipython-input-21-2b40773bcde1> in <module>
      7                      xaxis=None, yaxis=None, show_grid=False)
      8 tiles = EsriImagery.clone(crs=crs.GOOGLE_MERCATOR).opts(topts)
----> 9 taxi  = dd.read_parquet('../data/nyc_taxi_wide.parq').persist()
     10 colormaps = {n: palette[n] for n in ['fire','bgy','bgyw','bmy','gray','kbc']}

/opt/conda/lib/python3.7/site-packages/dask/base.py in persist(self, **kwargs)
    127         dask.base.persist
    128         """
--> 129         (result,) = persist(self, traverse=False, **kwargs)
    130         return result
    131 

/opt/conda/lib/python3.7/site-packages/dask/base.py in persist(*args, **kwargs)
    585         postpersists.append((rebuild, a_keys, state))
    586 
--> 587     results = schedule(dsk, keys, **kwargs)
    588     d = dict(zip(keys, results))
    589     results2 = [r({k: d[k] for k in ks}, *s) for r, ks, s in postpersists]

/opt/conda/lib/python3.7/site-packages/dask/threaded.py in get(dsk, result, cache, num_workers, pool, **kwargs)
     74     results = get_async(pool.apply_async, len(pool._pool), dsk, result,
     75                         cache=cache, get_id=_thread_get_id,
---> 76                         pack_exception=pack_exception, **kwargs)
     77 
     78     # Cleanup pools associated to dead threads

/opt/conda/lib/python3.7/site-packages/dask/local.py in get_async(apply_async, num_workers, dsk, result, cache, get_id, rerun_exceptions_locally, pack_exception, raise_exception, callbacks, dumps, loads, **kwargs)
    460                         _execute_task(task, data)  # Re-execute locally
    461                     else:
--> 462                         raise_exception(exc, tb)
    463                 res, worker_id = loads(res_info)
    464                 state['cache'][key] = res

/opt/conda/lib/python3.7/site-packages/dask/compatibility.py in reraise(exc, tb)
    110         if exc.__traceback__ is not tb:
    111             raise exc.with_traceback(tb)
--> 112         raise exc
    113 
    114     import pickle as cPickle

/opt/conda/lib/python3.7/site-packages/dask/local.py in execute_task(key, task_info, dumps, loads, get_id, pack_exception)
    228     try:
    229         task, data = loads(task_info)
--> 230         result = _execute_task(task, data)
    231         id = get_id()
    232         result = dumps((result, id))

/opt/conda/lib/python3.7/site-packages/dask/core.py in _execute_task(arg, cache, dsk)
    117         func, args = arg[0], arg[1:]
    118         args2 = [_execute_task(a, cache) for a in args]
--> 119         return func(*args2)
    120     elif not ishashable(arg):
    121         return arg

/opt/conda/lib/python3.7/site-packages/dask/dataframe/io/parquet.py in _read_pyarrow_parquet_piece(fs, piece, columns, index_cols, is_series, partitions, categories)
    901                            partitions=partitions,
    902                            use_pandas_metadata=True,
--> 903                            file=f)
    904 
    905     if pa.__version__ < LooseVersion('0.9.0'):

/opt/conda/lib/python3.7/site-packages/pyarrow/parquet.py in read(self, columns, use_threads, partitions, open_file_func, file, use_pandas_metadata)
    562             table = reader.read_row_group(self.row_group, **options)
    563         else:
--> 564             table = reader.read(**options)
    565 
    566         if len(self.partition_keys) > 0:

/opt/conda/lib/python3.7/site-packages/pyarrow/parquet.py in read(self, columns, use_threads, use_pandas_metadata)
    212             columns, use_pandas_metadata=use_pandas_metadata)
    213         return self.reader.read_all(column_indices=column_indices,
--> 214                                     use_threads=use_threads)
    215 
    216     def scan_contents(self, columns=None, batch_size=65536):

/opt/conda/lib/python3.7/site-packages/pyarrow/_parquet.pyx in pyarrow._parquet.ParquetReader.read_all()

/opt/conda/lib/python3.7/site-packages/pyarrow/error.pxi in pyarrow.lib.check_status()

ArrowIOError: Unknown encoding type.
```

### PR DESCRIPTION
**Pull request by jsignell**
- don't use gv for tiling
 - only include in geo deps
 - update other deps
 - always specify engine when using read_parquet (closes #178)
 - changing html for exercises so that it works in jlab

### CODE REVIEW

**Comment by jsignell** [comment]
Subsumes #176

File: examples/tutorial/01_Workflow_Introduction.ipynb [hunk 1]
```diff
@@ -711,7 +711,9 @@
     }
    },
    "source": [
-    "As you can see, you can specify geo plots easily with GeoViews, and if your HoloViews objects are too big to visualize in a browser directly, you can add `datashade()` to render them into images dynamically on zooming, etc.\n",
+    "As you can see, you can specify geo plots easily and if your HoloViews objects are too big to visualize in a browser directly, you can add `datashade()` to render them into images dynamically on zooming, etc.\n",
+    "\n",
+    "**NOTES:** If your plots require a different projection or shapefiles, you can use GeoViews to access lots of geographic functionality. See the notebook on [Geographic Data](07_Geographic_Data.ipynb) for more information.\n",
```
Comment (jbednar) [inline examples/tutorial/01_Workflow_Introduction.ipynb]:
```suggestion
    "**NOTE:** HoloViews includes support for basic web-based map tiles as used here, but if you need to work flexibly with different geographic projections, you'll want to install [GeoViews](https://geoviews.org) as well. See the notebook on [Geographic Data](07_Geographic_Data.ipynb) for more information.\n",
```

File: examples/tutorial/07_Geographic_Data.ipynb [hunk 2]
```diff
@@ -24,7 +24,9 @@
     "<a href=\"http://bokeh.pydata.org\"><img style=\"margin:8px; display:inline; object-fit:scale-down; max-height:150px\" src=\"../assets/bokeh.png\"/></a><br>\n",
     "<a href=\"http://scitools.org.uk/cartopy\"><img style=\"margin:8px; display:inline; object-fit:scale-down; max-height:120px\" src=\"../assets/cartopy.png\"/></a>\n",
     "<a href=\"http://matplotlib.org\"><img style=\"margin:4px; display:inline; object-fit:scale-down; max-height:150px\" src=\"../assets/matplotlib_wm.png\"/></a>\n",
-    "</div>"
+    "</div>\n",
+    "\n",
+    "**NOTE:** If you haven't installed geoviews yet, you can do so now using `conda install geoviews`. From the notebook type: `!conda install -y geoviews` in a fresh cell and run the cell. "
```
Comment (jbednar) [inline examples/tutorial/07_Geographic_Data.ipynb]:
```suggestion
    "**NOTE:** This notebook requires GeoViews. If you haven't installed it yet, you can do so now using `conda install geoviews`. From the notebook, type: `!conda install -y geoviews` in a fresh cell and run the cell. "
```

File: examples/tutorial/10_Working_with_Large_Datasets.ipynb [hunk 3]
```diff
@@ -179,7 +177,7 @@
    "source": [
     "## Adding a tile source\n",
     "\n",
-    "Using the GeoViews (geographic) extension for HoloViews, we can display a map in the background. Just declare a Bokeh WMTSTileSource and pass it to the gv.WMTS Element, then we can overlay it:"
+    "Using a publicly available [tiling service](https://en.wikipedia.org/wiki/Tile_Map_Service), we can display a map in the background. "
```
Comment (jbednar) [inline examples/tutorial/10_Working_with_Large_Datasets.ipynb]:
```suggestion
    "Using a publicly available [tiled map service](https://en.wikipedia.org/wiki/Tiled_web_map), we can display a geographic map in the background. "
```

File: examples/tutorial/exercises/Exercise-4-dynamic-interactions.ipynb [hunk 4]
```diff
@@ -348,7 +355,9 @@
    "source": [
     "As before, you'll need to complete the callback function so it returns an element.  In this case, we need a <code>Curve</code> plotting the 'fare_amount' against the 'tpep_pickup_datetime', starting something like:\n",
     "\n",
-    "<code>def fare_curve(data):\n",
+    "\n",
+    "```python\n",
+    "def fare_curve(data):\n",
     "    ...\n",
     "</code>"
```
Comment (jbednar) [inline examples/tutorial/exercises/Exercise-4-dynamic-interactions.ipynb]:
```suggestion
    "```"
```

**Review by jbednar — APPROVED** [review]
Looks great, thanks!

**Comment by jsignell** [comment]
The exercises solution rendering is still messed up, but once https://github.com/pyviz/nbsite/pull/135 is in, the links will at least point to the exercises. I made the notebook download links be at the top as well, so hopefully that will be good enough. I think it is an nbconvert bug that is making the rendered solutions be so terrible :(

**Pull request merged**